packages/strapi/lib/Strapi.js

```diff
@@ -341,15 +341,14 @@ class Strapi {
     this.models['strapi_webhooks'] = webhookModel(this.config);
 
     this.db = createDatabaseManager(this);
-
-    await this.runLifecyclesFunctions(LIFECYCLES.REGISTER);
-    await this.db.initialize();
-
     this.store = createCoreStore({
       environment: this.config.environment,
       db: this.db,
     });
 
+    await this.runLifecyclesFunctions(LIFECYCLES.REGISTER);
```

Comment:
this shouldn't be moved.

Reply:
It would be helpful to put the `createCoreStore` above t`his.db.initialize()` in order to have the coreStore available in the migration functions.
In the initialize, the corestore model is instanciated before the migration, so it should not break anything.
What do you think ?

Reply:
If we leave the code as is it means a user can use strapi.store inside a register function before it was initialized. the lifecycles must be before this init for sure. 

for the db.init you aren't going to make a change to use strapi.store in the migrations in the PR so I still wouldn't change the order here